### PR TITLE
[RFC][NI] Update ember version and upgrade to have its own environment

### PR DESCRIPTION
**Pull request by AndreJoaquim**
NI

### CODE REVIEW

File: config/environment.js [hunk 1]
```diff
@@ -1,6 +1,16 @@
 /*jshint node:true*/
+/* global module */
 'use strict';
 
 module.exports = function(/* environment, appConfig */) {
-  return { };
+  var ENV = {};
```
Comment (raffo0707) [inline config/environment.js:6]: why two steps?

File: lib/tasks/read-config.js [hunk 2]
```diff
@@ -12,27 +12,37 @@ module.exports = Task.extend({
...
+    if (!existsSync(this.parentConfigPath)) {
+      throw new Error(`Environment config does not exist at ${this.parentConfigPath}`);
     }
   },
 
-  _loadSpreadsheetConfig: function(superConfig) {
-    superConfig._result.credentials = require(`${this.project.root}/${superConfig._result.spreadsheet_cred_file}`);
+  _loadSpreadsheetConfig: function(parentConfig) {
+    if (Object.keys(parentConfig._result).length <= 0) {
+      throw new Error(`The parent config directory was not found.`)
```
Comment (raffo0707) [inline lib/tasks/read-config.js]: not concatenating anything, you can use single quote: ` => '

File: lib/tasks/read-config.js [hunk 3]
```diff
@@ -12,27 +12,37 @@ module.exports = Task.extend({
...
+    }
+
+    let config = parentConfig._result.uniTranslations;
+    let path = `${this.project.root}/${config.spreadsheetCredentialsFile}`;
+    if (!existsSync(path)) {
+        throw new Error(`There is no such credentials file.`)
```
Comment (raffo0707) [inline lib/tasks/read-config.js]: same

File: lib/commands/get.js [hunk 4]
```diff
@@ -1,46 +1,67 @@
...
 
-    let ReadConfigTask = require('../tasks/read-config');
-    let readConfig = new ReadConfigTask({project: this.project});
+    let readConfig = new ReadConfigTask({ project: this.project });
 
     readConfig.run().then(config => {
-      if (!config.locales) {
-        throw new SilentError('Missing loacales array in environment configuration.');
+      let locales = config.uniTranslations.locales;
+      if (!locales.length <= 0) {
```
Comment (raffo0707) [inline lib/commands/get.js]: empty line after let

File: lib/commands/get.js [hunk 5]
```diff
@@ -1,46 +1,67 @@
...
+  },
 
-    return `export default ${prettyTranslsation};`;
+  _rows: function(tab) {
+    return new Promise((resolve, reject) => {
+      return tab.getRows({offset: 0, limit: tab.rowCount}, (err, rows) => {
+        if (err) return reject(err);
+
+        let translations = this._initTranslations(this.locales, rows);
+        this._save(translations);
```
Comment (raffo0707) [inline lib/commands/get.js]: empty line after let

File: lib/commands/set.js [hunk 6]
```diff
@@ -14,21 +17,56 @@ module.exports = {
...
     this._super(...arguments);
 
-    let ReadConfigTask = require('../tasks/read-config');
     let readConfig = new ReadConfigTask({project: this.project});
 
     readConfig.run().then(config => {
-      if (!config.locales) {
-        throw new SilentError('Missing loacales array in environment configuration.');
+      let locales = config.uniTranslations.locales;
+      if (!locales.length <= 0) {
```
Comment (raffo0707) [inline lib/commands/set.js]: empty line after let

File: lib/commands/get.js [hunk 7]
```diff
@@ -1,46 +1,67 @@
+/* global module, require */
 'use strict';
 
+let ReadConfigTask = require('../tasks/read-config');
 let filesystem = require('../utils/filesystem');
+let Promise = require('ember-cli/lib/ext/promise');
```
Comment (raffo0707) [inline lib/commands/get.js]: I know was me that did the error first. But doing the salesforce integration I learned that what get from the require should be const and not let. would you mind to change it please? thanks 🙏

File: lib/tasks/read-config.js [hunk 8]
```diff
@@ -12,27 +12,37 @@ module.exports = Task.extend({
...
-  _loadSpreadsheetConfig: function(superConfig) {
-    superConfig._result.credentials = require(`${this.project.root}/${superConfig._result.spreadsheet_cred_file}`);
+  _loadSpreadsheetConfig: function(parentConfig) {
+    if (Object.keys(parentConfig._result).length <= 0) {
+      throw new Error(`The parent config directory was not found.`)
+    }
+
+    let config = parentConfig._result.uniTranslations;
+    let path = `${this.project.root}/${config.spreadsheetCredentialsFile}`;
+    if (!existsSync(path)) {
```
Comment (raffo0707) [inline lib/tasks/read-config.js]: empty line after let

**Review by raffo0707 — CHANGES_REQUESTED** [review]

File: config/environment.js (hunk 1 above)
Comment (AndreJoaquim) [reply config/environment.js:6]: While this was not necessary, it allows to reproduce better a default ember app code-styling (e.g. [environment.js](https://github.com/uniplaces/self-service-frontend/blob/dev/config/environment.js) in the self-service platform).

File: lib/commands/get.js (hunk 7 above)
Comment (AndreJoaquim) [reply lib/commands/get.js]: Of course :)

File: config/environment.js (hunk 1 above)
Comment (raffo0707) [reply config/environment.js:6]: top :) thanks for replying

**Review by raffo0707 — APPROVED** [review]

**Pull request merged**